Mathlib/Data/Complex/Basic.lean

```diff
@@ -578,16 +649,16 @@ theorem im_sq_le_normSq (z : ℂ) : z.im * z.im ≤ normSq z :=
 #align complex.im_sq_le_norm_sq Complex.im_sq_le_normSq
 
 theorem mul_conj (z : ℂ) : z * conj z = normSq z :=
-  ext_iff.2 <| by simp [norm_sq, mul_comm, sub_eq_neg_add, add_comm]
+  ext_iff.2 <| by simp [normSq, mul_comm, sub_eq_neg_add, add_comm, ofReal']
 #align complex.mul_conj Complex.mul_conj
 
 theorem add_conj (z : ℂ) : z + conj z = (2 * z.re : ℝ) :=
-  ext_iff.2 <| by simp [two_mul]
+  ext_iff.2 <| by simp [two_mul, ofReal']
 #align complex.add_conj Complex.add_conj
 
 /-- The coercion `ℝ → ℂ` as a `ring_hom`. -/
 def ofReal : ℝ →+* ℂ :=
-  ⟨coe, of_real_one, of_real_mul, of_real_zero, of_real_add⟩
+  ⟨⟨⟨fun (x: ℝ) ↦ (↑x : ℂ), ofReal_one⟩, ofReal_mul⟩, ofReal_zero, ofReal_add⟩
```

Comment:
Can you use `where` syntax here instead?

Reply:
Thanks. That is much better.